pkg/verifier/index_issue.go

```diff
@@ -52,78 +52,94 @@ func (IndexKnownIssues) VerifyRepair(ctx Context, idMatcher func(ulid.ULID) bool
 			}
 		}()
 
-		if err = objstore.DownloadFile(ctx, ctx.Logger, ctx.Bkt, path.Join(id.String(), block.IndexFilename), filepath.Join(tmpdir, block.IndexFilename)); err != nil {
-			return errors.Wrapf(err, "download index file %s", path.Join(id.String(), block.IndexFilename))
-		}
-
-		stats, err := block.GatherIndexHealthStats(ctx.Logger, filepath.Join(tmpdir, block.IndexFilename), meta.MinTime, meta.MaxTime)
-		if err != nil {
-			return errors.Wrapf(err, "gather index issues %s", id)
-		}
-
-		level.Debug(ctx.Logger).Log("stats", fmt.Sprintf("%+v", stats), "id", id)
-		if err = stats.AnyErr(); err == nil {
-			return nil
+		stats, err := verifyIndex(ctx, id, tmpdir, meta)
+		if err == nil {
+			level.Debug(ctx.Logger).Log("msg", "no issue", "id", id)
+			continue
 		}
 
 		level.Warn(ctx.Logger).Log("msg", "detected issue", "id", id, "err", err)
 
 		if !repair {
 			// Only verify.
-			return nil
+			continue
 		}
 
-		if stats.OutOfOrderChunks > stats.DuplicatedChunks {
-			level.Warn(ctx.Logger).Log("msg", "detected overlaps are not entirely by duplicated chunks. We are able to repair only duplicates", "id", id)
+		// Verify repaired block before uploading it.
```

Comment:
This comment does not make sense here does it?

Reply:
Here we just skip the repair step, but we need to continue to verify each index from the list we parsed.

Reply:
I agree the name of the function is clear enough.